src/Build/BackEnd/Components/SdkResolution/SdkResolution.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.BackEnd
+{
+    /// <summary>
+    ///     Component responsible for resolving an SDK to a file path. Loads and coordinates
+    ///     with <see cref="SdkResolver" /> plug-ins.
+    /// </summary>
+    internal class SdkResolution
+    {
+        private readonly object _lockObject = new object();
+        private readonly SdkResolverLoadingStrategy _sdkResolverLoadingStrategy;
+        private IList<SdkResolver> _resolvers;
+
+        /// <summary>
+        ///     Create an instance with a specified resolver assembly loading strategy. Used
+        ///     for testing purposes.
+        /// </summary>
+        /// <param name="sdkResolverLoadingStrategy">Resolver loading strategy.</param>
+        internal SdkResolution(SdkResolverLoadingStrategy sdkResolverLoadingStrategy)
+        {
+            _sdkResolverLoadingStrategy = sdkResolverLoadingStrategy;
+        }
+
+        internal static SdkResolution Instance { get; } = new SdkResolution(new SdkResolverLoadingStrategy());
+
+        /// <summary>
+        ///     Get path on disk to the referenced SDK.
+        /// </summary>
+        /// <param name="sdk">SDK referenced by the Project.</param>
+        /// <param name="logger">Logging service.</param>
+        /// <param name="buildEventContext">Build event context for logging.</param>
+        /// <param name="importElementLocation">Location of the element within the project which referenced the SDK.</param>
```

Comment:
importElementLocation makes it sound like SDKs can only come from msbuild import elements. Maybe SDKReferenceLocation would be a better name?